File: app/src/main/java/com/irishinterest/irishinterest/model/user/UserRegisterMessageEncrypted.java
package com.irishinterest.irishinterest.model.user;

import com.fasterxml.jackson.annotation.JsonAnyGetter;
import com.fasterxml.jackson.annotation.JsonAnySetter;
import com.fasterxml.jackson.annotation.JsonIgnore;
import com.fasterxml.jackson.annotation.JsonInclude;
import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.annotation.JsonPropertyOrder;

import java.util.HashMap;
import java.util.Map;

@JsonInclude(JsonInclude.Include.NON_NULL)
@JsonPropertyOrder({
        "firstname",
        "lastname",
        "publishername",
        "telephone",
        "position",
        "email",
        "password",
        "confirm",
        "usertype"
})
public class UserRegisterMessageEncrypted {

    @JsonProperty("firstname")
    private String firstname;
    @JsonProperty("lastname")
    private String lastname;
    @JsonProperty("publishername")
    private String publishername;
    @JsonProperty("telephone")
    private String telephone;
    @JsonProperty("position")
    private String position;
    @JsonProperty("email")
    private String email;
    @JsonProperty("password")
    private String password;
    @JsonProperty("confirm")
    private String confirm;
    @JsonProperty("usertype")
    private String usertype;
    @JsonIgnore
    private Map<String, Object> additionalProperties = new HashMap<String, Object>();

    @JsonProperty("firstname")
    public String getFirstname() {
        return firstname;
    }

    @JsonProperty("firstname")
    public void setFirstname(String firstname) {
        this.firstname = firstname;
    }

    @JsonProperty("lastname")
    public String getLastname() {
        return lastname;
    }

    @JsonProperty("lastname")
    public void setLastname(String lastname) {
        this.lastname = lastname;
    }

    @JsonProperty("publishername")
    public String getPublishername() {
        return publishername;
    }

    @JsonProperty("publishername")
    public void setPublishername(String publishername) {
        this.publishername = publishername;
    }

    @JsonProperty("telephone")
    public String getTelephone() {
        return telephone;
    }

    @JsonProperty("telephone")
    public void setTelephone(String telephone) {
        this.telephone = telephone;
    }

    @JsonProperty("position")
    public String getPosition() {
        return position;
    }

    @JsonProperty("position")
    public void setPosition(String position) {
        this.position = position;
    }

    @JsonProperty("email")
    public String getEmail() {
        return email;
    }

    @JsonProperty("email")
    public void setEmail(String email) {
        this.email = email;
    }

    @JsonProperty("password")
    public String getPassword() {
        return password;
    }

    @JsonProperty("password")
    public void setPassword(String password) {
        this.password = password;
    }

    @JsonProperty("confirm")
    public String getConfirm() {
        return confirm;
    }

    @JsonProperty("confirm")
    public void setConfirm(String confirm) {
        this.confirm = confirm;
    }

    @JsonProperty("usertype")
    public String getUsertype() {
        return usertype;
    }

    @JsonProperty("usertype")
    public void setUsertype(String usertype) {
        this.usertype = usertype;
    }

    @JsonAnyGetter
    public Map<String, Object> getAdditionalProperties() {
        return this.additionalProperties;
    }

    @JsonAnySetter
    public void setAdditionalProperty(String name, Object value) {
        this.additionalProperties.put(name, value);
    }

}

File: app/src/main/java/com/irishinterest/irishinterest/model/user/UserRegisterMessageEncrypted.kt
package com.irishinterest.irishinterest.model.user

import com.fasterxml.jackson.annotation.JsonInclude
import com.fasterxml.jackson.annotation.JsonPropertyOrder
import com.fasterxml.jackson.annotation.JsonProperty
import com.fasterxml.jackson.annotation.JsonIgnore
import com.fasterxml.jackson.annotation.JsonAnyGetter
import com.fasterxml.jackson.annotation.JsonAnySetter
import java.util.HashMap

@JsonInclude(JsonInclude.Include.NON_NULL)
@JsonPropertyOrder(
    "firstname",
    "lastname",
    "publishername",
    "telephone",
    "position",
    "email",
    "password",
    "confirm",
    "usertype"
)
class UserRegisterMessageEncrypted {
    @get:JsonProperty("firstname")
    @set:JsonProperty("firstname")
    @JsonProperty("firstname")
    var firstname: String? = null

    @get:JsonProperty("lastname")
    @set:JsonProperty("lastname")
    @JsonProperty("lastname")
    var lastname: String? = null

    @get:JsonProperty("publishername")
    @set:JsonProperty("publishername")
    @JsonProperty("publishername")
    var publishername: String? = null

    @get:JsonProperty("telephone")
    @set:JsonProperty("telephone")
    @JsonProperty("telephone")
    var telephone: String? = null

    @get:JsonProperty("position")
    @set:JsonProperty("position")
    @JsonProperty("position")
    var position: String? = null

    @get:JsonProperty("email")
    @set:JsonProperty("email")
    @JsonProperty("email")
    var email: String? = null

    @get:JsonProperty("password")
    @set:JsonProperty("password")
    @JsonProperty("password")
    var password: String? = null

    @get:JsonProperty("confirm")
    @set:JsonProperty("confirm")
    @JsonProperty("confirm")
    var confirm: String? = null

    @get:JsonProperty("usertype")
    @set:JsonProperty("usertype")
    @JsonProperty("usertype")
    var usertype: String? = null

    @JsonIgnore
    private val additionalProperties: MutableMap<String, Any> = HashMap()
    @JsonAnyGetter
    fun getAdditionalProperties(): Map<String, Any> {
        return additionalProperties
    }

    @JsonAnySetter
    fun setAdditionalProperty(name: String, value: Any) {
        additionalProperties[name] = value
    }
}